engine-tests/src/tests/sanity.rs

```diff
@@ -694,6 +694,27 @@ fn test_eth_transfer_incorrect_nonce() {
     utils::validate_address_balance_and_nonce(&runner, dest_address, Wei::zero(), 0.into());
 }
 
+#[test]
+fn test_shanghai_contract() {
+    let (mut runner, mut source_account, _) = initialize_transfer();
+
+    // Panicked with `EvmFatal(Other("internal error: entered unreachable code [engine/src/engine.rs:148:54]"))`
+    let result = runner
+        .submit_with_signer(&mut source_account, |nonce| {
+            aurora_engine_transactions::legacy::TransactionLegacy {
+                nonce,
+                gas_price: 0.into(),
+                gas_limit: u64::MAX.into(),
+                to: None,
+                value: Wei::zero(),
+                data: hex::decode("6080604052348015600e575f80fd5b50607480601a5f395ff3fe6080604052348015600e575f80fd5b50600436106026575f3560e01c8063919840ad14602a575b5f80fd5b600560405190815260200160405180910390f3fea2646970667358221220cb01b9b9c75e5cd079a1980af2fe4397d2029888d12737d74cbbc10e0de65bd364736f6c63430008150033").unwrap(),
+            }
+        })
+        .unwrap();
+
+    println!("{result:?}");
```

Comment:
Println not needed, could think about adding an assert on the output instead or something.